apps/teams-test-app/src/public/externalOauth_end.html

```diff
@@ -1,29 +1,42 @@
-<!DOCTYPE html>
 <html>
   <head>
+    <script
+      type="text/javascript"
+      src="
+    https://res.cdn.office.net/teams-js/2.21.0/js/MicrosoftTeams.min.js"
+    ></script>
     <h2 style="font-size: 40px">Starting mock external oauth2</h2>
-  </head>
-  <body>
     <script>
-      const failureMessage = {
-        type: 'authenticationFailed',
-        data: 'error',
-      };
-      const successMessage = {
-        type: 'authenticationSuccess',
-        data: 'success',
-      };
+      function appendLog(msg) {
+        const element = document.createElement('p');
+        element.textContent = msg;
+        document.getElementById('log').appendChild(element);
+      }
 
       const params = new URLSearchParams(window.location.search);
       const titleId = params.get('titleId');
 
-      if (window.opener) {
-        if (titleId === 'testOauthFailureTitleId') {
-          window.opener.postMessage(failureMessage, '*');
+      async function callteamsJS() {
+        if (microsoftTeams) {
+          appendLog('succesfuuly loaded microsoftTeams');
```

Comment:
>succesfuuly [](http://example.com/codeflow?start=21&length=11)

Don't change this now because I don't want to delay things further but maybe come back later and fix this misspelling.

Reply:
oh no. It just rerun for 1 min. Not a big deal. I will fix it.